### ISSUE TITLE
rootSearchGraph returning wrong number of results in metadata

### ISSUE THREAD

**Issue by berger-n** [buda-base/lds-pdi#172]
in http://purl.bdrc.io/lib/rootSearchGraph?LG_NAME=zh-hant&I_LIM=500&L_NAME=%22%E7%84%A1%E7%95%8F%E7%AC%88%E5%A4%9A%22&format=json
there is only one person returned
but metadata say there are 2 of them

**Comment by MarcAgate** [comment]
There are actually much more than 2 people ! Run the following query to get the results:

```
CONSTRUCT
  {
    ?TT a ?type .
    ?TT ?pp ?lit .
    ?TT skos:prefLabel ?l.
    ?TT tmp:matchScore ?sc .
    ?TT tmp:prefLabelMatch ?lit1.
    ?TT tmp:altLabelMatch ?lit2.
    
    ?TT bdo:workType ?wT .

    ?TT bdo:workPartType ?wPT.
    ?TT bdo:workHasPart ?wHP.
    ?TT bdo:workExpressionOf ?wE.
    ?TT tmp:prefLabelExpressionOf ?litE.
    ?TT ?wE ?litE.
    #?TT tmp:prefLabelExpressionOfMatch ?litEH.
    ?TT bdo:workHasExpression ?wH.
    ?TT tmp:prefLabelHasExpression ?litH.
    ?TT ?wH ?litH.
    #?TT tmp:prefLabelHasExpressionMatch ?litHH.

    #?TT bdo:workTranslationOf ?wTr.
    #?TT bdo:workHasTranslation ?wHTr.

    ?TT ?eventType ?lit .

    ?TT tmp:creatorLabel ?litC .
    ?TT tmp:creatorRole ?roleLabel .

    ?TT owl:sameAs ?same .
    ?TT ?same ?sameL .
    ?TT owl:sameAs ?sameBDRC .
    ?TT ?sameBDRC ?sameBDRClabel .
    ?TT a ?sameT .
    ?TT adm:canonicalHtml ?sameCanonHTML .
    ?TT adm:canonicalHtml ?canonHTML .

    ?TT skos:prefLabel ?sameL.
    ?TT skos:prefLabel ?sameBDRClabel.

    ?TT adm:status ?status .

  }
WHERE {

  {
    (?s ?sc ?lit) text:query ( rdfs:label "無畏笈多"@zh-hant "highlight:" ).
    {
      ?TT ?anyprop ?anypropobj .       
      ?anypropobj :eventWho ?s ; a ?eventType .
    } 
    union 
    {
      ?TT ?p ?s.
      FILTER(!EXISTS { ?TT :eventWho ?anywho })
      ?s rdf:type ?pp .
    }

  }
  union
  { 
    (?s ?sc ?litC) text:query ( skos:prefLabel "無畏笈多"@zh-hant "highlight:").
    ?TT bdo:creator ?agent .
    ?agent bdo:agent ?s ; bdo:role ?creaRole .
    ?creaRole skos:prefLabel ?roleLabel . 
  }
  union
  {
    (?TT ?sc ?lit2) text:query ( skos:altLabel "無畏笈多"@zh-hant "highlight:" ).
  }
  union
  {
    (?TT ?sc ?lit1) text:query ( skos:prefLabel "無畏笈多"@zh-hant "highlight:" ).
  }

  ?TT rdf:type ?type .  

  OPTIONAL{?TT skos:prefLabel ?l.}.

  optional { ?TT adm:canonicalHtml ?canonHTML } .

  optional { 
    ?TT bdo:workPartOf* ?res .
    ?adm adm:adminAbout ?res ; adm:status ?status . 
  }

  optional
  { 
    ?TT owl:sameAs ?same .
    ?TT a ?sameT . 
    ?sameT rdfs:subClassOf+ bdo:Entity . 

    OPTIONAL{ ?same skos:prefLabel ?sameL . } . 
    OPTIONAL{ ?same foaf:name ?sameL . } . 
    optional { ?same adm:canonicalHtml ?sameCanonHTML } .
  }

  optional
  { 
    ?sameBDRC owl:sameAs+ ?TT ; owl:sameAs ?same .
    ?sameBDRC a ?sameT . 
    ?sameT rdfs:subClassOf+ bdo:Entity . 
    OPTIONAL{ ?sameBDRC skos:prefLabel ?sameBDRClabel . } .      
    
    OPTIONAL{ ?same skos:prefLabel ?sameL . } . 
    OPTIONAL{ ?same foaf:name ?sameL . } . 
    optional { ?same adm:canonicalHtml ?sameCanonHTML } .
  }

  OPTIONAL{?TT bdo:workPartType ?wPT. }
  OPTIONAL{?TT bdo:workHasPart ?wHP. }
  #?TT bdo:workType bdr:WorkTypeAbstractWork.}.
  OPTIONAL{
    ?TT bdo:workExpressionOf ?wE.
    ?wE skos:prefLabel ?litE.
    #OPTIONAL{
    #  (?wE ?scE ?litEH) text:query ( skos:prefLabel ?L_NAME "highlight:").
    #}
   }.
  OPTIONAL{
    ?TT bdo:workHasExpression ?wH.
    ?wH skos:prefLabel ?litH.
    #OPTIONAL{
    #   (?wH ?scH ?litHH) text:query ( skos:prefLabel ?L_NAME "highlight:").
    #}
  }.
 # Good idea but translation property is totally unused... yet ?
 # OPTIONAL{
 #   ?TT bdo:workTranslationOf ?wTr.
 #   ?wTr skos:prefLabel ?litTr.
 #   OPTIONAL{
 #     (?wTr ?scTr ?litTrH) text:query ( skos:prefLabel ?L_NAME "highlight:").
 #   }
 #  }.
 #  OPTIONAL{
 #   ?TT bdo:workHasTranslation ?wHTr.
 #   ?wHTr skos:prefLabel ?litHTr.
 #   OPTIONAL{
 #     (?wHTr ?scHTr ?litHTrH) text:query ( skos:prefLabel ?L_NAME "highlight:").
 #   }
 # }.
}
order by desc(?sc)
```

And note that the sole person resource that is seen in the json result has this weird form: 

```
dila:A015944  a             bdo:Person ;
        tmp:matchScore      "24.503796"^^xsd:float ;
        tmp:prefLabelMatch  "↦無畏笈多↤"@zh-hant ;
        bdr:P4004           "'jigs med 'byung gnas sbas pa/"@bo-x-ewts ;
        dila:A015944        "Abhayākaragupta"@sa-x-iast , "無畏笈多"@zh-hant , "अभयाकरगुप्त"@und-Zyyy ;
        owl:sameAs          dila:A015944 , bdr:P4004 ;
        skos:prefLabel      "'jigs med 'byung gnas sbas pa/"@bo-x-ewts , "Abhayākaragupta"@sa-x-iast , "無畏笈多"@zh-hant , "अभयाकरगुप्त"@und-Zyyy .
```
that translates in the json as :

```
{
xml:lang: "sa-x-iast",
type: "http://purl.dila.edu.tw/resource/A015944",
value: "Abhayākaragupta"
},
{
xml:lang: "zh-hant",
type: "http://purl.dila.edu.tw/resource/A015944",
value: "無畏笈多"
},
{
xml:lang: "und-Zyyy",
type: "http://purl.dila.edu.tw/resource/A015944",
value: "अभयाकरगुप्त"
},
```

**where the predicate is the URI of the resource, not teh URI of a property of it.**

This affects the parsing of the results.

Also, check the following:

```
<http://purl.dila.edu.tw/resource/A009300>
        a                   bdo:Person ;
        tmp:altLabelMatch   "阿婆↦多↤"@zh-hant ;
        tmp:matchScore      "4.204282"^^xsd:float , "5.1892"^^xsd:float ;
        tmp:prefLabelMatch  "阿羅婆↦多↤尊者"@zh-hant ;
        skos:prefLabel      "阿羅婆多尊者"@zh-hant .
```

and note that we have a altLabelMatch that pertains to only one character when L_NAME="無畏笈多" - That looks like a lucene issue with  zh-hant

Moreover, the presence of triplets such as:

dila:A015944 dila:A015944 "Abhayākaragupta"@sa-x-iast , "無畏笈多"@zh-hant , "अभयाकरगुप्त"@und-Zyyy ; 

might indicate some issue in DILA data integration.

WDYT ?

**Comment by berger-n** [comment]
as we just skyped about it,
```
{
  xml:lang: "sa-x-iast",
  type: "http://purl.dila.edu.tw/resource/A015944",
  value: "Abhayākaragupta"
}
```
is a way I have found to be able to make pairs among returned subvalues for a given search result
see http://library.bdrc.io/search?q=%22bodhicary%C4%81vat%C4%81ra%22&lg=sa-x-iast&t=Any
it allows us to display titles of `hasExpression` works instead of their ID as it used to:
![image](https://user-images.githubusercontent.com/33383857/68683830-c26e4b80-0567-11ea-9246-f221c6fbece0.png)

**Comment by MarcAgate** [comment]
Fixed as of commit 097e518 .

**Comment by berger-n** [comment]
thanks!
